src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,316 @@
+                <Tag
+                  defaultActive={index === 0}
+                  id={`${props.id}-option-${props.optionIDGetter(getOption(val))}`}
+                  key={props.optionIDGetter(getOption(val))}
+                  onRemove={onOptionRemove}
+                  readOnly={props.readOnly}
+                  ref={valueRefs[index]}
+                >
+                  {props.optionLabelGetter(
+                    props.options.find(option => props.optionIDGetter(option) === props.optionIDGetter(getOption(val))),
+                  )}
+                </Tag>
+              ))
+            }
+            <input
+              aria-autocomplete="list"
+              aria-controls={ids.listbox}
+              aria-describedby={`${ids.emptyMessage}`}
+              aria-expanded={expanded}
+              aria-haspopup="listbox"
+              aria-labelledby={ids.label}
+              autoComplete="off"
+              role="combobox"
+              className={classNames.input}
+              id={ids.textbox}
+              onBlur={onAutocompleteBlur}
+              onChange={onAutocompleteChange}
+              placeholder={value.length === 0 ? props.placeholder : undefined}
+              readOnly={props.readOnly}
+              required={props.required ? value.length === 0 : false}
+              ref={autocompleteRef}
+              value={autocompleteValue}
+            />
+          </TagList>
+          <FormControlIcons validationMessage={props.validationMessage} className={classNames.iconsContainer}>
+            {value.length > 0 && (
+              <IconButton
+                icon={iconClear}
+                label={`Remove all selected options on ${props.label}`}
+                onPress={onOptionsClear}
+                className={classNames.iconClear}
+              />
+            )}
+            <Icon icon={iconCaretDown} label="Open multi select options" />
+          </FormControlIcons>
+        </div>
+        {dropdownContents()}
+      </FormControl>
+    </div>
+  );
+});
+
+MultiSelect.propTypes = {
+  classNames: PropTypes.shape({
+    container: PropTypes.string,
+    formControlChildrenContainer: PropTypes.string,
+    input: PropTypes.string,
+    tagList: PropTypes.string,
+  }),
+  id: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  listboxChildren: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
+  name: PropTypes.string.isRequired,
+  onChange: PropTypes.func,
+  options: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.arrayOf(PropTypes.string)]).isRequired,
+  optionIDGetter: PropTypes.func,
+  optionLabelGetter: PropTypes.func,
+  placeholder: PropTypes.string,
+  readOnly: PropTypes.bool,
+  required: PropTypes.bool,
+  showLoader: PropTypes.bool, // eslint-disable-line react/no-unused-prop-types
+  tagChildren: PropTypes.func,
+  validationMessage: PropTypes.string,
+  value: PropTypes.arrayOf(PropTypes.string),
```

Comment:
What about the usages of `value` being a list of objects?

Reply:
Oooooh, this is interesting! I definitely meant to backfill a change to this when I implemented the object handling behavior and forgot. What makes it interesting is that I get no prop-type errors in the console for the object example in MDS site or the tests...